src/land/smap9km_ssm2ioda.py

```diff
@@ -0,0 +1,173 @@
+            self.varAttrs[iodavar, iconv.OvalName()]['units'] = 'm3 m-3'
+            self.varAttrs[iodavar, iconv.OerrName()]['units'] = 'm3 m-3'
+            self.varAttrs[iodavar, iconv.OqcName()]['units'] = 'm3 m-3'
+
+        # open input file name
+        ncd = nc.Dataset(self.filename, 'r')
+        # set and get global attributes
+        self.satellite = "SMAP"
+        self.sensor = "radar and radiometer"
+        AttrData["platform"] = self.satellite
+        AttrData["sensor"] = self.sensor
+
+        data = ncd.groups['Soil_Moisture_Retrieval_Data'].variables['soil_moisture'][:]
+        vals = data[:].ravel()
+        _FillValue = ncd.groups['Soil_Moisture_Retrieval_Data'].variables['soil_moisture'].getncattr('_FillValue')
+        valid_max = ncd.groups['Soil_Moisture_Retrieval_Data'].variables['soil_moisture'].getncattr('valid_max')
+        valid_min = ncd.groups['Soil_Moisture_Retrieval_Data'].variables['soil_moisture'].getncattr('valid_min')
+
+        lats = ncd.groups['Soil_Moisture_Retrieval_Data'].variables['latitude'][:].ravel()
+        lons = ncd.groups['Soil_Moisture_Retrieval_Data'].variables['longitude'][:].ravel()
+        errs = ncd.groups['Soil_Moisture_Retrieval_Data'].variables['soil_moisture_error'][:].ravel()
+        qflg = ncd.groups['Soil_Moisture_Retrieval_Data'].variables['retrieval_qual_flag'][:].ravel()
+        sflg = ncd.groups['Soil_Moisture_Retrieval_Data'].variables['surface_flag'][:].ravel()
+        vegop = ncd.groups['Soil_Moisture_Retrieval_Data'].variables['vegetation_opacity'][:].ravel()
+
+        times = np.empty_like(vals, dtype=object)
+
+        if self.mask == "maskout":
+            with np.errstate(invalid='ignore'):
+                mask = (vals > valid_min) & (vals < valid_max)
+            vals = vals[mask]
+            lats = lats[mask]
+            lons = lons[mask]
+            errs = errs[mask]
+            qflg = qflg[mask]
+            sflg = sflg[mask]
+            vegop = vegop[mask]
+            times = times[mask]
+
+        # get datetime from filename
+        str_split = self.filename.split("_")
+        str_datetime = str_split[7]
+        my_datetime = datetime.strptime(str_datetime, "%Y%m%dT%H%M%S")
+        base_datetime = my_datetime.strftime('%Y-%m-%dT%H:%M:%SZ')
+        vals = vals.astype('float32')
+        lats = lats.astype('float32')
+        lons = lons.astype('float32')
+        errs = errs.astype('float32')
+        qflg = qflg.astype('int32')
+        sflg = sflg.astype('int32')
+        vegop = vegop.astype('int32')
+
+        for i in range(len(lons)):
+            times[i] = base_datetime
+            errs[i] = 0.04
+        # add metadata variables
+        self.outdata[('dateTime', 'MetaData')] = times
+        self.outdata[('latitude', 'MetaData')] = lats
+        self.outdata[('longitude', 'MetaData')] = lons
+        self.varAttrs[('dateTime', 'MetaData')]['units'] = 'seconds since 2021-12-21'
```

Comment:
seconds since 2021-12-21???

Reply:
This is what used in ioda-validate for units check.  I think that I can change it as what SMAP data decided. I will check with Ryan @rhoneyager.

Reply:
There is some confusion with the dateTime variable. There are currently two datetime formats for ioda:
- ISO 8601 string value
    - These values are held in a string variable
- epoch style with a reference date time (epoch) and a time offset in seconds
    - These values are held in a 64-bit integer variable
    - The epoch is defined in the units attribute "seconds since my_epoch" where my_epoch is an ISO 8601 datetime value

The string style will eventually get deprecated since it has serious performance issues.

To keep these two datetime representations straight, the ioda reader expects the string style to be in the variable "MetaData/datetime" and the epoch style in the variable "MetaData/dateTime" (note the upper case "T" in dateTime).

You've got the string style in the "MetaData/dateTime" variable which is confusing the validator (and will confuse the ioda reader). You need to either change the variable name to "MetaData/datetime" or convert the times variable (strings) to an epoch with an offset in seconds. It's probably easiest to switch the variable to "MetaData/datetime" at this point.